node/actors/bft/src/testonly/ut_harness.rs

```diff
@@ -71,26 +71,46 @@ impl UTHarness {
         }
     }
 
-    pub(crate) fn check_recovery_after_timeout(&mut self) {
-        self.set_replica_view(self.owner_as_view_leader_next());
-
-        let base_replica_view = self.replica_view();
-        let base_leader_view = self.leader_view();
-        assert!(base_leader_view < base_replica_view);
+    pub(crate) async fn iterate_next(&mut self) {
+        let leader_commit = self.new_procedural_leader_commit_many().await;
+        self.dispatch_leader_commit(leader_commit).await.unwrap();
+        self.recv_signed()
+            .await
+            .unwrap()
+            .cast::<ReplicaPrepare>()
+            .unwrap();
+    }
 
-        assert_eq!(self.replica_phase(), Phase::Prepare);
+    pub(crate) async fn check_recovery_after_timeout(
+        &mut self,
+        view: ViewNumber,
+        high_vote_view: ViewNumber,
+        high_qc_view: ViewNumber,
+    ) {
```

Comment:
Personally I don't understand this function's signature. What are the arguments for? Can you make it simpler? And/or please document when exactly should it be called.

Reply:
Docs added.